Ziffity/AddCustomer/Console/Command/CreateCustomers.php

```diff
@@ -0,0 +1,170 @@
+        $this->csvImport = $csvImport;
+        $this->jsonImport = $jsonImport;
+
+    }
+    /**
+     * @return void
+     */
+    public function configure(): void
+    {
+        $this->setName("customer:importer");
+        $this->setDescription("Import customers data from csv or json file");
+        $this->addOption(
+            'profile',
+            null,
+            InputOption::VALUE_REQUIRED,
+            'Profile Name only supports csv and json formats'
+        );
+        $this->addArgument('source', InputArgument::REQUIRED, 'File Path');
+        parent::configure();
+
+    }
+
+    /**
+     * @param InputInterface $input
+     * @param OutputInterface $output
+     * @return int|null
+     */
+    public function execute(InputInterface $input, OutputInterface $output): ?int
+    {
+        try {
+            $source = $input->getArgument('source');
+            $profile = $input->getOption('profile');
+            $varDirectory = $this->directoryList->getPath('var');
+            $magentoRootDirectory = $this->directoryList->getRoot();
+           //Validate File Type
+            $pathInfo = $this->fileSystemIo->getPathInfo($source);
+            $extension = $pathInfo['extension'] ?? '';
+            if (!$profile) {
+                $profile = $extension;
+            }
+            if ((!($profile === "csv" || $profile === "json")) || ($profile !== $extension)) {
+                throw new Exception(__("Invalid file type!"));
+            }
+            //Validate File Size
+            $maxImageSize = $this->fileSize->getMaxFileSizeInMb();
+            if ($maxImageSize) {
+                $message = __('The total size of the uploadable files can\'t be more than %1M', $maxImageSize);
+            } else {
+                $message = __('System doesn\'t allow to get file upload settings');
+            }
+            $profileObj = $this->getProfileObj($profile);
+
+            $sourcePath = $magentoRootDirectory . '/' . $source;
+            $destinationPath = $varDirectory . '/' . $source;
+            $this->fileSystemIo->cp($sourcePath, $destinationPath);
+            $this->state->setAreaCode(Area::AREA_GLOBAL);
+            $output->writeln('<info>Execution starts.</info>');
+            //Importing customers
+            $this->customer->install($destinationPath, $output,$profileObj,$this->progressBarFactory);
+            $output->write(PHP_EOL);
+            $output->writeln('<info>Execution ends.</info>');
+            return Cli::RETURN_SUCCESS;
+        } catch (Exception $e) {
+            $msg = $e->getMessage();
+            $output->writeln("<error>$msg</error>", OutputInterface::OUTPUT_NORMAL);
+            return Cli::RETURN_FAILURE;
+        }
+    }
+
+    /**
+     * @param $profile
+     * @return void
+     */
+    private function getProfileObj($profile) {
+        switch ($profile) {
+            case "csv":
+                return $this->csvImport;
+            case "json":
```

Comment:
@msakila532 declare as constant and use the same in cases.

Reply:
Update the constant value